client/src/Pos/Client/Txp/Util.hs

```diff
@@ -373,20 +373,22 @@ makeLenses ''GroupedInputPickerState
 type GroupedInputPicker = StateT GroupedInputPickerState (Either TxError)
 
 groupedInputPicker :: InputPickingWay
-groupedInputPicker utxo outputs moneyToSpent =
-    evalStateT (pickInputs []) (GroupedInputPickerState moneyToSpent sortedGroups)
+groupedInputPicker utxo = \outputs moneyToSpent ->
+    let outputAddrsSet =
+            foldl' (flip HS.insert) mempty $
+            map (txOutAddress . toaOut) outputs
+        isOutputAddr = flip HS.member outputAddrsSet
+        preparedGroups = filter (not . isOutputAddr . ugAddr) sortedGroups
+        disallowedInputGroups = filter (isOutputAddr . ugAddr) gUtxo
```

Comment:
Can use `partition` here to avoid double iterating the list, and then only have to sort the smaller set.

Reply:
Right, didn't notice this thing.

I didn't get point about
> have to sort the smaller set

thought. I have to filter after sorting because filtering requires `outputs` be known.

Reply:
``` sortedGroups = sortOn (Down . ugTotalMoney) . filter (not . isOutputAddr . ugAddr ) $ gUtxo```

Reply:
The point of this PR is to do sorting once when `groupedInputPicker`, with fed `utxo`, is executed multiple times;
in ^ way I will get `filter` part executed when `outputs` is given.